SteamKit2/SteamKit2/Steam/WebAPI/WebAPI.cs

```diff
@@ -242,11 +242,11 @@ public async Task<KeyValue> CallAsync( HttpMethod method, string func, int versi
                     paramBuilder.Append( "/?" ); // start our GET params
                 }
 
-                args.Add( "format", "vdf" );
+                args[ "format" ] = "vdf";
```

Comment:
Isn't the better fix to not set that if it already exists instead of silently overriding?

Reply:
@xPaw if user already set something else, such as `format: json` then we'll fail on parsing. I believe it's a correct way to go "this MUST be vdf, regardless what it is right now".

The steam API key could be improved indeed in a way that we check if key is set and if not we don't add our own, but I don't know if it's appropriate, it's a matter of what we trust more - key in the provided args, or key in the iface. Right now it's key in the iface, with adnotation of "if you don't want to override it then don't set it yourself first".

Reply:
We could go with very advanced logic of "if it's already set to something else than vdf then throw, if it's already vdf then do nothing, if it's not then set it as vdf" but I'm not sure if we really want to go that deep, maybe I'll give it a try to satisfy the first case without a need of deep-cloning the dictionary.

Reply:
Considering this method always parseskeyvalues for you, I'd say passing in `format` is always unwanted, regardless of its value.

Reply:
Yes but SK2 sets it itself and I'll crash with it if I throw on each occurance of format, it's no different from current code which tries to add it and crashes on its existence.

Reply:
I took that thought into account anyway, so the function doesn't crash only if `format` either doesn't exist or is already `vdf`, doesn't matter whether the user or SK2 set it.